### PR TITLE
MultiText dataset Added to Benchmarking

### PR DESCRIPTION
**Pull request by wistuba**
MultiText dataset in #354 enabled for benchmarking and documented. Some cleanup and bug fixes as well.

Added new argument for benchmarking script that allows to easily change between running Joint with and without reset.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: benchmarks/run_benchmark.py [hunk 1]
```diff
@@ -59,6 +59,10 @@ def load_config(scenario_file, model_file, updater_file, dataset_file):
         default=5 * 24 * 3600,
         help="Maximum execution time.",
     )
+    parser.add_argument(
+        "--no-reset", action="store_true", help="Do not reset model weights (Joint and GDumb only)"
```
Comment (prabhuteja12) [inline benchmarks/run_benchmark.py:63]: Not important: can you make it more descriptive? `--use-prior-task-weights` ?

File: src/renate/benchmark/datasets/nlp_datasets.py [hunk 2]
```diff
@@ -142,14 +142,13 @@ def tokenize_fn(batch):
 class MultiTextDataModule(DataIncrementalDataModule):
     """
     Inspired by the dataset used in "Episodic Memory in Lifelong Language Learning"
-    by d’Autume et al. this is a collection of five different datasets that we call domains:
+    by d’Autume et al. this is a collection of four different datasets that we call domains:
```
Comment (prabhuteja12) [inline src/renate/benchmark/datasets/nlp_datasets.py:145]: Why the change?

File: src/renate/benchmark/datasets/nlp_datasets.py [hunk 3]
```diff
@@ -142,14 +142,13 @@ def tokenize_fn(batch):
...
     AGNews, Yelp, DBPedia and Yahoo Answers.
 
     The output space if the union of the output space of all the domains.
     The dataset has 33 classes: 4 from AGNews, 5 from Yelp, 14 from DBPedia, and 10 from Yahoo.
 
     The maximum allowed size for the training set is 115000 and for the test set is 7600.
```
Comment (prabhuteja12) [inline src/renate/benchmark/datasets/nlp_datasets.py:151]: What does 'allowed' mean here?

File: src/renate/benchmark/datasets/nlp_datasets.py (hunk 2 above)
Comment (wistuba) [reply src/renate/benchmark/datasets/nlp_datasets.py:145]: originally, it was a collection of 5 datasets. after amazon review was removed, it dropped to 4.

File: src/renate/benchmark/datasets/nlp_datasets.py (hunk 3 above)
Comment (wistuba) [reply src/renate/benchmark/datasets/nlp_datasets.py:151]: anything larger will raise an exception. i assume the smallest dataset has a size of 115k train or 7600 test points.

any suggestion how to improve the description to explain it better?

**Review by prabhuteja12 — APPROVED** [review]
